src/utils/ar_data_price.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Immutable data container class representing a market price in Winston for a particular volume
+ * of data that enforces valid number ranges for byte counts and Winston price values.
+ */
+export class ARDataPrice {
+	/**
+	 * @returns an ARDataPrice instance with the given byte count and Winston amount
+	 * @throws {@link Error} if negative or non-integer values are provided for either value
+	 */
+	constructor(readonly numBytes: number, readonly winstonPrice: number) {
+		if (numBytes < 0 || !Number.isInteger(numBytes) || winstonPrice < 0 || !Number.isInteger(winstonPrice)) {
+			throw new Error(
+				`numBytes (${numBytes}) and winstonPrice (${winstonPrice}) should be non-negative integer values.`
+			);
+		}
+
+		this.numBytes = numBytes;
+		this.winstonPrice = winstonPrice;
```

Comment:
I don't understand why the linter doesn't complain here about writing a read-only attribute hmmm...
I think these lines don't actually take any effect

Reply:
I didn't know that :o 

Anyway, we are setting it to whatever it's already set; we set exactly the same value, and that is the same than doing nothing. I think...